content/post/2017-05-30-databases-and-sql.Rmd

```diff
@@ -389,85 +400,94 @@ result <- dbSendQuery(dbcon, query)
 
-https://cran.r-project.org/web/packages/dplyr/vignettes/databases.html
+Here is an example of its use:
 
 ```{r dplyr-dbcon}
 library(dplyr)
 
-bety_src <- src_postgres(dbname = "bety", password = 'bety', host = 'terra-bety.default', user = 'bety', port = 5432)
-
+bety_src <- src_postgres(dbname = "bety",
+                        password = 'bety',
+                        host = 'terra-bety.default',
+                        user = 'bety',
+                        port = 5432)
 
 #translate_sql()
 ```
 
 ```{r complex-query}
-treatments <- tbl(bety_src, 'treatments') %>% 
+treatments <- tbl(bety_src, 'treatments') %>%
   select(treatment_id = id , name, definition, control)
 
 managements_treatments <- tbl(bety_src, 'managements_treatments')
 
-managements <- tbl(bety_src, 'managements') %>% 
-  filter(mgmttype %in% c('Fertilization_N', 'Planting', 'Irrigation')) %>% 
-  select(management_id = id, date, mgmttype, level, units) %>% 
-  left_join(managements_treatments, by = 'management_id') %>% 
+managements <- tbl(bety_src, 'managements') %>%
+  filter(mgmttype %in% c('Fertilization_N', 'Planting', 'Irrigation')) %>%
+  select(management_id = id, date, mgmttype, level, units) %>%
+  left_join(managements_treatments, by = 'management_id') %>%
   left_join(treatments, by = 'treatment_id')
 
-planting <-managements %>% 
-  filter(mgmttype == "Planting") %>% 
+planting <-managements %>%
+  filter(mgmttype == "Planting") %>%
   select(treatment_id, planting_date = date, nrate = level)
 
-heights <- tbl(bety_src, 'traits_and_yields_view') %>% 
-  filter(trait %in% c('plant_height', 'canopy_height')) %>% 
+heights <- tbl(bety_src, 'traits_and_yields_view') %>%
+  filter(trait %in% c('plant_height', 'canopy_height')) %>%
   left_join(planting, by = 'treatment_id') %>%  
-  collect %>% 
-  mutate(age = as.Date(raw_date) - planting_date) 
+  collect %>%
+  mutate(age = as.Date(raw_date) - planting_date)
 
 library(ggplot2)
 ggplot(data = heights, aes(raw_date, mean)) +
-  geom_point(alpha = 0.1) + 
+  geom_point(alpha = 0.1) +
   facet_wrap(~trait, scales = 'free')
 ```
 
 ### APIs
 
+The data at teraref can also be accessed via an API:
+
 * sign into terraref.ncsa.illinois.edu/bety
   * username: `pi4-student`
-  * password: `ask-instructor` <!--pi4student!!-->
-* locate your API key and copy it <!-- sSJzqPF77rRus9BGOZS4T76fzNhQ5Nk01tjcf1Bh-->
+  * password: `ask-instructor`
+
+* locate your API key and copy it
 
 * see [terraref tutorials 01- betydb getting started](https://github.com/terraref/tutorials/blob/master/traits/00-BETYdb-getting-started.Rmd)
+
 * API Documentation terraref.ncsa.illinois.edu/bety/api/beta/docs
 
 #### restful interface
 
-get put delete 
+get put delete
 https://en.wikipedia.org/wiki/Representational_state_transfer
 
 #### json data structures
```

Comment:
What do we want to say here about json? Do we have some examples that we looked at in class?